stdlib/sql/awsathena.go

```diff
@@ -0,0 +1,197 @@
+			columnPointers[i] = &m.columns[i]
+		}
+		if err := m.Cursor.Scan(columnPointers...); err != nil {
+			return false
+		}
+	}
+	return next
+}
+
+func (m *AwsAthenaRowReader) GetNextRow() ([]values.Value, error) {
+	row := make([]values.Value, len(m.columns))
+	for i, column := range m.columns {
+		switch value := column.(type) {
+		case bool, int64, float64:
+			row[i] = values.New(value)
+		case int:
+			row[i] = values.NewInt(int64(value))
+		case int8:
+			row[i] = values.NewInt(int64(value))
+		case int16:
+			row[i] = values.NewInt(int64(value))
+		case int32:
+			row[i] = values.NewInt(int64(value))
+		case float32:
+			row[i] = values.NewFloat(float64(value))
+		case string:
+			row[i] = values.New(value)
+		case time.Time:
+			switch m.sqlTypes[i].DatabaseTypeName() {
+			case "date":
+				row[i] = values.NewString(value.Format(layoutDate))
+			case "time":
+				row[i] = values.NewString(value.Format(layoutTime))
+			case "timestamp":
+				row[i] = values.NewString(value.Format(layoutTimeStampNtz))
+			default:
+				row[i] = values.NewTime(values.ConvertTime(value))
+			}
+		case nil:
+			row[i] = values.NewNull(flux.SemanticType(m.columnTypes[i]))
+		default:
+			execute.PanicUnknownType(flux.TInvalid)
+		}
+	}
+	return row, nil
+}
+
+func (m *AwsAthenaRowReader) InitColumnNames(names []string) {
+	m.columnNames = names
+}
+
+func (m *AwsAthenaRowReader) InitColumnTypes(types []*sql.ColumnType) {
+	fluxTypes := make([]flux.ColType, len(types))
+	for i := 0; i < len(types); i++ {
+		switch types[i].DatabaseTypeName() {
+		case "tinyint", "smallint", "int", "integer", "bigint":
+			fluxTypes[i] = flux.TInt
+		case "float", "double", "real":
+			fluxTypes[i] = flux.TFloat
+		case "boolean":
+			fluxTypes[i] = flux.TBool
+		case "timestamp with time zone": // "timestamp", "date" and "time" will be represented as string
+			fluxTypes[i] = flux.TTime
+		default:
+			fluxTypes[i] = flux.TString
+		}
+	}
+	m.columnTypes = fluxTypes
+	m.sqlTypes = types
+}
+
+func (m *AwsAthenaRowReader) ColumnNames() []string {
+	return m.columnNames
+}
+
+func (m *AwsAthenaRowReader) ColumnTypes() []flux.ColType {
+	return m.columnTypes
+}
+
+func (m *AwsAthenaRowReader) SetColumnTypes(types []flux.ColType) {
+	m.columnTypes = types
+}
+
+func (m *AwsAthenaRowReader) SetColumns(i []interface{}) {
+	m.columns = i
+}
+
+func (m *AwsAthenaRowReader) Close() error {
+	if m.CloseFunc != nil {
+		return m.CloseFunc()
+	}
+	if err := m.Cursor.Err(); err != nil {
+		return err
+	}
+	return m.Cursor.Close()
+}
+
+func NewAwsAthenaRowReader(r *sql.Rows) (execute.RowReader, error) {
+	reader := &AwsAthenaRowReader{
+		Cursor: r,
+	}
+	cols, err := r.Columns()
+	if err != nil {
+		return nil, err
+	}
+	reader.InitColumnNames(cols)
+
+	types, err := r.ColumnTypes()
+	if err != nil {
+		return nil, err
+	}
+	reader.InitColumnTypes(types)
+
+	return reader, nil
+}
+
+// This is code for `sql.to`, which is not supported now (or yet).
```

Comment:
I'd prefer if we didn't leave commented out source code in the tree.

Reply:
You're right. Removed.